app/src/main/java/org/sagebionetworks/research/mpower/EntryActivity.kt

```diff
@@ -32,12 +32,41 @@
 
 package org.sagebionetworks.research.mpower
 
+import android.Manifest
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Build
 import android.os.Bundle
-import androidx.fragment.app.FragmentManager
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import com.google.android.gms.location.ActivityRecognition
+import com.google.android.gms.location.ActivityTransition
 import dagger.android.support.DaggerAppCompatActivity
-
+import org.sagebionetworks.research.mpower.viewmodel.PassiveGaitViewModel
+import org.slf4j.LoggerFactory
 
 class EntryActivity : DaggerAppCompatActivity() {
+
+    private val LOGGER = LoggerFactory.getLogger(EntryActivity::class.java)
+
+    private val passiveGaitViewModel: PassiveGaitViewModel by lazy {
+        ViewModelProvider(this).get(PassiveGaitViewModel::class.java)
+    }
+
+    private val trackingTransitionsObserver = Observer<Boolean> { tracking ->

```

Comment:
i could move enable/disable stuff to a utility class.  I would definitely not move any ViewModels or observers into Application though.  I want the lifecycle of those objects to be bound to the activity so that it gets cleaned up as soon as activity is killed by system.  EntryActivity is the correct place to wire all of this up.